src/btree/bt_curnext.c

```diff
@@ -124,9 +124,18 @@ __cursor_fix_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
         cbt->v = __bit_getv_recno(cbt->ref, cbt->recno, btree->bitcnt);
         cbt->iface.value.data = &cbt->v;
     } else {
-        if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DISK) && upd->type != WT_UPDATE_TOMBSTONE)
-            return (__wt_value_return(cbt, upd));
-        cbt->iface.value.data = upd->data;
+        if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DISK)) {
+            switch (upd->type) {
+            case WT_UPDATE_TOMBSTONE:
+                cbt->iface.value.data = upd->data;
+                __wt_free_update_list(session, &upd);
+                break;
+            default:
+                return (__wt_value_return(cbt, upd));
+            }
+        }
+        if (upd != NULL)
+            cbt->iface.value.data = upd->data;
     }
     cbt->iface.value.size = 1;
```

Comment:
This is not related to your change specifically but this looks weird to me.

If we just pointed our cursor to an update's buffer, shouldn't we be copying the size across to instead of just making it 1? If `WT_UPDATE_RESTORED_FROM_DISK` is set, `__wt_value_return` will do the right thing. But the code path where we select an update from the in-memory update list looks suspicious to me. It seems like we'd only get the first byte of the update's value.

Reply:
@tetsuo-cpp Fixed length column store has data size of 1 :)

Reply:
I didn't know that. Thanks!